src/xtgeo/grid3d/_gridprop_roxapi.py

```diff
@@ -98,17 +96,20 @@ def _convert_to_xtgeo_prop(
 
     mybuffer[iind, jind, kind] = pvalues[cellno]
 
-    if self._isdiscrete:
+    if result["discrete"]:
         mybuffer = ma.masked_greater(mybuffer, xtgeo.UNDEF_INT_LIMIT)
     else:
         mybuffer = ma.masked_greater(mybuffer, xtgeo.UNDEF_LIMIT)
 
-    self._values = mybuffer
-    self._name = pname
+    result["values"] = mybuffer
+    result["name"] = pname
 
-    if self._isdiscrete:
-        self.codes = _fix_codes(self, roxprop.code_names)
-        logger.info("Fixed codes: %s", self.codes)
+    if result["discrete"]:
+        result["codes"] = _fix_codes(
+            result["values"].reshape(-1).compressed(), roxprop.code_names
+        )
+        logger.info("Fixed codes: %s", result["codes"])
+    return result
 
 
 def export_prop_roxapi(
```

Comment:
Remove self-parameter

Reply:
This is the export functions so they do indeed take an object instance. Whether it should be named self is another thing. I think that renaming is out of scope, but could be made a separate issue.